### PR TITLE
feat: show market status and winning outcome on cardbet

### LINKED ISSUE #126: Invalid market state not considered in `My Bets`
For example, this market https://presagio.pages.dev/markets?id=0x14d1c2c828ba13da17f1a38fb9b67ef7853ec711 is finalizing as Invalid.

Our rebel agent bet on `Yes` outcome because it was cheap (price <0.5), so after the market is resolved as invalid, he indeed will be profitable from this bet (because in the end, `Yes` token is worth 0.5 after market resolves as invalid).

However, in my bets section, it looks like the market doesn't have an answer yet and that he lost the whole investment:

Screenshot by Dropbox Capture

### PR DESCRIPTION
**Pull request by berteotti**
image

Example profile: https://feat-pgio-107-show-outcome-s.presagio.pages.dev/profile?address=0x91fEA319Ed45c7b12683c8c25e741151fBFee24b

Closes #126

### CODE REVIEW

File: app/components/CardBet.tsx [hunk 1]
```diff
@@ -83,42 +83,71 @@ export const CardBet = ({ userBets, children }: CardBetProps) => {
...
+              )}
             </div>
           </div>
         </section>
       </Link>
-      <section className="flex items-center border-t border-outline-base-em px-4 py-2 md:h-[48px] md:py-0">
+      <section className="flex items-center border-t border-outline-base-em p-4 md:h-[48px]">
         <div className="flex w-full items-center justify-between space-x-4">
-          <div className="flex flex-col items-start space-y-0.5 md:flex-row md:items-center md:space-x-3">
+          <div className="flex flex-col items-start space-y-1.5 md:flex-row md:items-center md:space-x-3 md:space-y-0">
```
Comment (Diogomartf) [inline app/components/CardBet.tsx:145]: we probably can use gap here, so we don't need to md:space-y-0. 
Also I think we should increase the space x on md:, hence the `md:gap-5`

```suggestion
          <div className="flex flex-col items-start gap-y-1.5 md:flex-row md:items-center md:gap-5">
```

File: entities/markets/market.ts [hunk 2]
```diff
@@ -68,8 +69,24 @@ export class Market {
...
-    return this.isClosed && this.answer !== null && this.answer !== Market.INVALID_ANSWER
+    return this.isClosed && this.answer !== null && !this.isAnswerInvalid
       ? this.outcomes[this.answer]
       : null;
   }
+
+  getCurrentAnswerOutcome() {
+    return this.currentAnswer !== null && !this.isCurrentAnswerInvalid
+      ? this.outcomes[this.currentAnswer]
+      : null;
```
Comment (Diogomartf) [inline entities/markets/market.ts:80]: why we're checking if `isCurrentAnswerInvalid`, wouldn't we also expect to receive invalid if it is the case on `getCurrentAnswerOutcome()`?

Comment (Diogomartf) [reply entities/markets/market.ts:80]: not sure I understood

Comment (berteotti) [reply entities/markets/market.ts:80]: If answer is invalid we are giving it a -1 index. We can't get -1 index from outcomes array to represent an invalid outcome. Currently we are returning null. There is a possibility of using `.at` method to return `array.length - 1 ` outcome.

Comment (Diogomartf) [reply entities/markets/market.ts:80]: yeah, we can try that

Comment (berteotti) [reply entities/markets/market.ts]: Done

File: app/components/CardBet.tsx (hunk 1 above)
Comment (berteotti) [reply app/components/CardBet.tsx]: Used close values to ones you suggested

**Review by Diogomartf — APPROVED** [review]

File: app/(main)/markets/Bet.tsx [hunk 3]
```diff
@@ -11,20 +11,14 @@ import {
...
-import { Swapbox } from '@/app/components';
+import { Swapbox, tagSchemeColors } from '@/app/components';
 import { KLEROS_URL, REALITY_QUESTION_URL } from '@/constants';
 import { PropsWithChildren } from 'react';
 
 interface BetProps {
   fixedProductMarketMaker: FixedProductMarketMaker;
 }
 
-const tagSchemeColors: Record<number, TagProps['colorScheme']> = {
```
Comment (ElRodrigote) [inline app/(main)/markets/Bet.tsx:22]: We should remove the TagProps import

File: app/components/CardBet.tsx [hunk 4]
```diff
@@ -27,6 +27,12 @@ interface CardBetProps extends PropsWithChildren {
   userBets: UserBets;
 }
 
+export const tagSchemeColors: Record<number, TagProps['colorScheme']> = {
+  0: 'success',
+  1: 'danger',
+  [Market.INVALID_ANSWER]: 'quaternary',
+};
```
Comment (ElRodrigote) [inline app/components/CardBet.tsx:34]: Is it worth transforming this into a utils fn? 
We use a similar logic at:
`app/(main)/markets/MarketActivity.tsx`

The piece of code I'm talking about:
```tsx
const TAG_COLOR_SCHEMES: { 0: TagColorSchemeProp; 1: TagColorSchemeProp } = {
  0: 'success',
  1: 'danger',
};
```

**Review by ElRodrigote — APPROVED** [review]

File: app/components/CardBet.tsx (hunk 4 above)
Comment (berteotti) [reply app/components/CardBet.tsx:34]: I think an object is ok for this. Where should we declare it?

**Pull request merged**